src/ext/NetFx/ca/netfxca.cpp

```diff
@@ -449,6 +453,30 @@ static HRESULT FileIdExists(
     return hr;
 }
 
+/******************************************************************
+ GetNetCoreCheckBinaryId - gets NetCoreCheck key from Binary table
+
+********************************************************************/
+static HRESULT GetNetCoreCheckBinaryId(
+    __in LPCWSTR pwzPlatform,
+    __out LPWSTR* ppwzBinaryId
+    )
+{
+    HRESULT hr = S_OK;
+
+    if (0 == lstrcmpW(L"x86", pwzPlatform)) {
+        hr = StrAllocString(ppwzBinaryId, L"Wix4NetCheck_x86", 0);
+    } else if (0 == lstrcmpW(L"x64", pwzPlatform)) {
```

Comment:
`lstrcmpW` uses the user locale in the compare. In general, you'd want invariant compare `::CompareStringW(LOCALE_INVARIANT,`. And based on https://github.com/wixtoolset/issues/issues/6947 you might start the path towards `CompareStringOrdinal()` here.